Ossrh Migration Support

jeremiahjstacey opened the pull request:
Closes #891 

Contains pom configuration changes and documentation updates accounting for migration to OSSRH maven central hosting.

Documentation updates performed through google docs, exported as .odt and .pdf formats.

jeremiahjstacey committed:
Applying pom changes required for OSSRH Migration.

jeremiahjstacey committed:
Setting the deployment name to the project name to make our deployments easier to locate in the maven central repository.

jeremiahjstacey committed:
Appending the project version to the maven central deployment name.  Thought I could use the project name to create a customized filter URL to simplify the link in the instructions, but that doesn't appear to work as desired.  This qualifier should make it easier to identify the desired project contents much easier among multiple deployments.

jeremiahjstacey committed:
Updating release steps documentation for the OSSRH migration.

kwwall commented:
@jeremiahjstacey - Sorry about that... I made a few additional comments and some very minor corrections to https://docs.google.com/document/d/1ObOHGk9LzxMUdZH_1ONqzSJxl78zjGXzNkZB5iCmE-E/edit?tab=t.0 before I realized that you'd saved that as an ODT and PDF here.

I must say that you have done an exemplary job with the documentation though. I really appreciate you taking this over. Thanks again.

kwwall reviewed (approved):
LGTM. I didn't check, but I presume that this still uploads the  files used with
```xml
<classifier>jakarta</classifier>
```
such as "esapi-2._x.y.z_-jakarta.jar" file, right?

Also, some new comments over at Google Docs.

jeremiahjstacey commented:
> LGTM. I didn't check, but I presume that this still uploads the files used with
> 
> ```
> <classifier>jakarta</classifier>
> ```
> 
> such as "esapi-2._x.y.z_-jakarta.jar" file, right?

the jakarta artifacts were created and published, and should be publicly available
https://repo1.maven.org/maven2/org/owasp/esapi/esapi/2.7.0.1-RC1/

jeremiahjstacey committed:
Documentation - Release Steps Updates

Updating section 2.3 with sonatype-supported gpg key servers
Updating section 7 with new Reference URLs for updated workflow.

jeremiahjstacey commented:
> Also, some new comments over at Google Docs.

Updated section 7 (references) with some new link information.  Most of the Sonatype links were no longer valid:
<img width="300" height="150" alt="image" src="https://github.com/user-attachments/assets/ceebbba5-3b9b-4962-b2e4-86e04d8270e8" />

Note the **GPG Management**::[Sonatype Supported Servers](https://central.sonatype.org/publish/requirements/gpg/#distributing-your-public-keySonatype) content.  Servers listed in the Sonatype documentation have been added to section 2.3, and invalid links have been removed
<img width="300" height="183" alt="image" src="https://github.com/user-attachments/assets/aac126f3-be71-422e-a537-ba6425b08771" />

Thank you for calling out the invalid references!

xeno6696 reviewed (approved):
Looks like the documentation was much more difficult than the actual change here.
